monitoring/messagesigning/keys/mock_priv.pem

```diff
@@ -0,0 +1 @@
+Private Key for Mock Uss goes here
```

Comment:
A script to generate the keys would be required to allow the end to end testing.

Reply:
This is tricky. You can gen a pub/priv keypair with `pycryptodome` to use for testing, but these keys will not pass the FAA SCVP validation. Because of this, we asked FAA to give us mock keys that would pass their cert eval, they are located in `monitoring/messagesigning/keys`. This requires the least amount of work since by default these valid keys are going to be used. Based on other comments in this PR, we will add an endpoint to make the keypair to be used configurable, so that negative tests can be run etc, but no need to have to gen a keypair since a valid pair already exists.

Reply:
It would be great to complete #899 to get a good wholistic view of how each component PR will fit into the whole.  For instance, this is a question of what kind of test data will be provided where.  There are a number of different possibilities (including build/test-certs and monitoring/uss_qualifier/test_data) depending on the overall approach.  But, I don't think we've identified a reason why a `monitoring/messagesigning` folder needs to exist -- it seems like it was initially created separately from mock_uss because of [unfamiliarity with mock_uss architecture](https://github.com/interuss/dss/pull/897#discussion_r1022024581).  Using multiple workers and the SynchronizedValue tool used in other parts of mock_uss, this probably doesn't need to be a separate monitoring tool.  It would be great to discuss in #899 (see [architecture suggestion](https://github.com/interuss/dss/pull/899#discussion_r1023154837)).

Reply:
The idea behind making `messagesigning` it's own package in `monitoring` is it won't just be used by `mock_uss`. `mock_uss` uses it to sign responses and validate incoming request http signatures, but it's also used by `monitoring.monitorlib.infrastructure.AuthAdapter` to sign outgoing requests to the USS under test (to ensure that individual USSs are correctly validating request signatures). I will work on the requested changes for `mock_uss` but i still think `messagesigning` should be its own package since it's being shared across components.

Reply:
Linking the [PR](https://github.com/interuss/dss/pull/893) containing the `messagesigning` code

Reply:
The best place to put code shared across multiple pieces of monitoring infrastructure is the `monitoring/monitorlib` folder -- either in a single .py file if it's small enough, or, e.g., `monitoring/monitorlib/messagesigning`.  That still leaves the question of where test data should be placed and that would probably depend on what would use the test data and how.

`monitoring.monitorlib.infrastructure.AuthAdapter` (presumably the specific [`SignedRequest` implementation](https://github.com/interuss/dss/blob/37fbff5c070cd899a8e4e1e18ee9ff78def93a22/monitoring/monitorlib/auth.py#L252)) signs outgoing requests to an auth server, but not a USS under test; it would be helpful to clarify expectations, hopefully in #899.

I'm worried about linking to open PRs within a PR.  It's important to have [small and focused PRs](https://github.com/interuss/dss/blob/master/CONTRIBUTING.md#general-principles) and a small and focused PR should not depend on other work in progress.  Linking to another open PR within a PR seems to be suggesting that understanding this PR depends on understanding the linked PR, and that suggests that this PR is not small and focused.  PRs need to be reviewed (and updated, when necessary) and merged one at a time.  I would suggest that #899 is the best candidate for first PR to complete and merge.  #891 has a number of open comments, but updating the dummy auth API YAML does seem like a well-scoped set of changes for a single PR that wouldn't need a design document (like the one in #899).

Reply:
I will focus on #899. To clarify my above comment, I didn't mean `SignedRequest`, I meant the overall parent `AuthAdapter`. To add the required message signing headers, I had to edit the `add_headers` method. This means that all outgoing requests are signed technically, not just to the USS under test.

Reply:
Could you please clarify what you mean by test data? We're reusing test data that already exists. Its basically re-using already existing testing scenarios, and running message signing analysis on the requests/responses associated with that scenario.